src/provider/rpc.ts

```diff
@@ -90,14 +90,45 @@ export class RpcProvider implements ProviderInterface {
   }
 
   public async getBlock(blockIdentifier: BlockIdentifier = 'pending'): Promise<GetBlockResponse> {
-    const method =
-      typeof blockIdentifier === 'string' && isHex(blockIdentifier)
-        ? 'starknet_getBlockByHash'
-        : 'starknet_getBlockByNumber';
+    const method = 'starknet_getBlockWithTxHashes';
```

Comment:
I think this is more complex than it should be. I think we can remove these if-elses (since it's mostly copy-paste anyway) and have just one:  
``` js
return this.fetchEndpoint('starknet_getBlockWithTxHashes', ...).then(
        this.responseParser.parseGetBlockResponse
 );
```
... and let the client send block_hash, block_number... Btw shouldn't it be just `block_id` instead of  block_hash, block_number? Specification states only  `block_id` as the name